pkg/services/authz/zanzana.go

```diff
@@ -37,7 +44,42 @@ func ProvideZanzana(cfg *setting.Cfg, db db.DB, features featuremgmt.FeatureTogg
 	var client zanzana.Client
 	switch cfg.Zanzana.Mode {
 	case setting.ZanzanaModeClient:
-		conn, err := grpc.NewClient(cfg.Zanzana.Addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		tokenClient, err := authnlib.NewTokenExchangeClient(authnlib.TokenExchangeConfig{
+			Token:            cfg.Zanzana.Token,
+			TokenExchangeURL: cfg.Zanzana.TokenExchangeURL,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		authInterceptor := func(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+			stackID := cfg.StackID
+			if stackID == "" {
+				stackID = "default"
+			}
+
+			token, err := tokenClient.Exchange(ctx, authnlib.TokenExchangeRequest{
+				Namespace: fmt.Sprintf("stacks-%s", stackID),
+				Audiences: []string{zanzanaAudience},
+			})
+			if err != nil {
+				return err
+			}
+
+			md := metadata.Pairs()
+			md.Set(authnlib.DefaultAccessTokenMetadataKey, token.Token)
+			ctx = metadata.NewOutgoingContext(ctx, md)
+			return invoker(ctx, method, req, reply, cc, opts...)
+		}
+
+		dialOptions := []grpc.DialOption{
+			grpc.WithUnaryInterceptor(authInterceptor),
+		}
+		if cfg.Env == setting.Dev {
+			dialOptions = append(dialOptions, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
I'm not sure if there's any difference with default value, since we do not specify any `TransportCredentials` by default?